openpype/hosts/maya/plugins/publish/collect_arnold_scene_source.py

```diff
@@ -21,15 +21,11 @@ def process(self, instance):
             if members is None:
                 self.log.warning("Skipped empty instance: \"%s\" " % objset)
                 continue
-            if objset.endswith("content_SET"):
-                members = cmds.ls(members, long=True)
-                children = get_all_children(members)
-                instance.data["contentMembers"] = children
-                self.log.debug("content members: {}".format(children))
-            elif objset.endswith("proxy_SET"):
-                set_members = get_all_children(cmds.ls(members, long=True))
-                instance.data["proxy"] = set_members
-                self.log.debug("proxy members: {}".format(set_members))
+
+            instance.data["contentMembers"] = self.get_ass_data(
+                objset, members, "content_SET")
+            instance.data["proxy"] = self.get_ass_data(
+                objset, members, "proxy_SET")
```

Comment:
```suggestion
            if objset.endswith("content_SET"):
                instance.data["contentMembers"] = self.get_hierarchy(members)
            if objset.endswith("content_SET"):
                instance.data["proxy"] = self.get_hierarchy(members)
```

Reply:
I think this one is not related. I did a wrong branch unfortunately